pkg/scheduler/objects/application.go

```diff
@@ -961,11 +972,22 @@ func (sa *Application) tryAllocate(headRoom *resources.Resource, nodeIterator fu
 
 		iterator := nodeIterator()
 		if iterator != nil {
-			alloc := sa.tryNodes(request, iterator)
-			// have a candidate return it
-			if alloc != nil {
+			if alloc := sa.tryNodes(request, iterator); alloc != nil {
+				// have a candidate return it
 				return alloc
 			}
+
+			// no nodes qualify, attempt preemption
+			if *preemptAttemptsRemaining > 0 {
+				*preemptAttemptsRemaining--
+				fullIterator := fullNodeIterator()
+				if fullIterator != nil {
```

Comment:
When is `fullIterator == nil`? Is this a legit case? I don't see coverage in the unit tests.

Reply:
The node iterator will return nil if there are no nodes to iterate. This is simply defensive coding to guard against that case. DIdn't seem worthy of a unit test.